apps/loan/src/components/PageLlamaMarkets/cells/UtilizationCell.tsx

```diff
@@ -1,4 +1,18 @@
-import { LendingVaultFromApi } from '@/entities/vaults'
+import { LendingVault } from '@/entities/vaults'
+import { CellContext } from '@tanstack/react-table'
+import Box from '@mui/material/Box'
 
-export const UtilizationCell = ({ data }: { data: LendingVaultFromApi }) =>
-  (data.borrowed.usdTotal && (100 * data.totalSupplied.usdTotal) / data.borrowed.usdTotal).toFixed(2) + '%'
+export const UtilizationCell = ({ getValue }: CellContext<LendingVault, LendingVault['utilizationPercent']>) => {
+  const value = getValue()
+  if (value == null) {
+    return '-'
+  }
+  return (
+    <>
+      {value.toFixed(2) + '%'}
+      <Box sx={(t) => ({ border: `1px solid ${t.design.Color.Primary[600]}`, height: 16 })}>
+        <Box sx={(t) => ({ backgroundColor: `${t.design.Color.Primary[500]}`, width: `${value}%`, height: '100%' })} />
```

Comment:
Double `<Box>` confuses me, is it really necessary?

Reply:
One is for the border, the one inside is the filled part, say 50% width for 50% utilization

Reply:
I present you, modern CSS:
```tsx
<Box
  sx={(t) => ({
    height: 16,
    border: `1px solid ${t.design.Color.Primary[600]}`,
    background: `linear-gradient(to right, ${t.design.Color.Primary[500]} ${value}%, transparent ${value / 2}%)`,
  })}
/>
```

Reply:
I dislike how the percentage is also not part of the progress bar itself. I'll ask @0xtutti about it in Slack, but you can try:
```tsx
    <Box
      sx={(t) => ({
        height: 24,
        border: `1px solid ${t.design.Color.Primary[600]}`,
        background: `linear-gradient(to right, ${t.design.Color.Primary[500]} ${value}%, transparent ${value / 2}%)`,
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        color: 'white',
        textShadow: '0 0 4px rgba(0,0,0,0.8)',
      })}
    >
      {value.toFixed(2) + '%'}
    </Box>
```
![image](https://github.com/user-attachments/assets/b613d218-6a49-41ff-94e2-be097ce1a906)
![image](https://github.com/user-attachments/assets/e10f27ce-0455-43cc-aae9-36eb6302334b)
![image](https://github.com/user-attachments/assets/7443449f-6048-4131-96c5-7830e356374a)

Reply:
not sure the linear-gradient solution is simpler, but I like it :nerd_face:

Reply:
for the alternative design lets wait tutti